src/chapters/unit-testing/hello-jasmine.rst

```diff
@@ -0,0 +1,301 @@
+
+   A Jasmine project
+
+There are three important files:
+
+#. ``reverse.js`` contains the ``reverse`` function, which must be exported for use in other files.
+#. ``spec/reverse.spec.js`` contains the tests for ``reverse``.
+#. ``index.js`` contains the Jasmine code needed to run the tests. This is the file that executes when you hit *run* in repl.it.
+
+.. warning:: Jasmine can be set up and used in many different ways. If you are looking for an answer on the Internet,---on Stack Overflow or in the Jasmine documentation---you will see widely varying usages of Jasmine that don't apply to your situation. Rely on this book as your main reference, and you'll be fine.
+
+Hello, Jasmine!
+---------------
+
+Let's build a "Hello, World!" Jasmine project, to get familiar with the basic components. Open and fork `this repl.it project <https://repl.it/@launchcode/Hello-Jasmine-Starter-Code>`_
+
+We will walk you through the steps need to get a simple Jasmine project up and running. Code along with us throughout this section.
+
+
+``index.js``
+^^^^^^^^^^^^
+
+This is the main project file. Up until now, ``index.js`` is where you have been writing the code for a given exercise or assignment. Now that we are writing tests for our code, ``index.js`` will contain the Jasmine code to find and execute the tests. Our project-specific code will live in other files.
+
+.. sourcecode:: js
+   :linenos:
+
+   const Jasmine = require('jasmine');
+   const jasmine = new Jasmine();
+
+   jasmine.loadConfig({
+      spec_dir: 'spec',
+      spec_files: [
+         "**/*[sS]pec.js"
+      ],
+   });
+
+   jasmine.execute();
+
+There are three main components of this program:
+
+#. Lines 1-2 import the Jasmine module and create a new Jasmine object, ``jasmine``. This object is responsible for finding and executing our tests.
+#. Lines 4-9 configure Jasmine to look for tests in the ``spec/`` directory of our project. Any file in this directory of the form ``fileName.spec.js`` will be assumed to contain tests, and will be executed by Jasmine.
+#. Line 11 triggers Jasmine to find and execute the tests.
+
+.. admonition:: Try It!
+
+   Hit *run* on the project. Two things happen:
+
+   - repl.it installs Jasmine.
+   - Jasmine searches for tests, finding none.
+
+Let's add some code to test.
+
+``hello.js``
+^^^^^^^^^^^^
+
+Create a new file in your project, ``hello.js``. Add this code to the file:
+
+.. sourcecode:: js
+   :linenos:
+
+   function hello(name) {
+      if (name === undefined)
+         name = "World";
+
+      return "Hello, " + name + "!";
+   }
+
+The ``hello`` function takes a single argument representing a person's name and returns a string greeting that person. If the function is called without an argument, the function returns ``"Hello, World!"``.
+
+To use this function outside ``hello.js`` we must export it. Add this statement at the bottom of the file.
+
+.. sourcecode:: js
+   :linenos:
+
+   module.exports = hello;
```

Comment:
Remind me to update the exercises, studio and repl.it files to match this syntax.